### ISSUE TITLE
Consolidate to a Weapon Base.

### ISSUE THREAD

**Issue by ndo360** [TTTReborn/tttreborn#209]
Greetings!

I know that I am not a part of the development team but this is a topic I feel strongly about for a few reasons and I hope I could persuade the team to look into, and consider, consolidating all weapon development under a unified base for long term ease of development.

**Why is this needed?**
I feel this is needed mainly due to the sole reason that later on down the line I feel there would need to be tutorials made for designing custom weapons for the gamemode. Without a unified weapon base I feel the guide may run the risk of the weapons being too unique to draw a simplified guide.

**Custom or Existing Base?**
Honestly I feel like we could use an existing base made by community members and is one that is referenced on the S&Box wiki page. That being the [Simple Weapon Base](https://github.com/timmybo5/simple-weapon-base) made by timmybo5. The licenses are compatible and has various existing weapons to tinker with.

**Wait... These are CSS weapons...**
I am not saying to use these weapons, despite the port job being done phenomenally well, as they can be replaced with whatever weapon models the team desires.

**Closing Remarks**
I know this isn't likely on the forefront of the team's mind, which is fair, but I wanted this issue made so it can be considered once the team is ready to start adding in various weapons in bulk to the gamemode. I feel the simple weapon base will fit well for the gamemode, given how easy it is to tweak, but if the team desires they can make their own base

**Comment by mzegar** [comment]
I think for now we can leave this off until we start building out more weapons. I would like to see what FacePunch has planned down the road (if they end up changing things). For now, we have plenty of work that needs to be done outside of weapons, and can be ignored for a bit.

**Comment by mzegar** [comment]
Here is the new PR I opened regarding the SWB https://github.com/TTTReborn/tttreborn/pull/375

As I'm one of the only ones left slowly chipping away at this project, we'll be using SWB to handle weapons. It's pretty well done, and I don't think it's wise to reinvent the wheel with the limited time I have for this project.

**Comment by mzegar** [comment]
Spoke with @Alf21 , he is not interested in using SWB, closing. I'm considering my own fork at the moment with SWB implemented.
